internal/mage/sdk/python.go

```diff
@@ -0,0 +1,87 @@
+package sdk
+
+import (
+	"context"
+	"os"
+
+	"dagger.io/dagger"
+	"github.com/dagger/dagger/internal/mage/util"
+	"github.com/magefile/mage/mg"
+)
+
+var _ SDK = Go{}
+
+type Python mg.Namespace
+
+// Lint lints the Python SDK
+func (t Python) Lint(ctx context.Context) error {
+	c, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	_, err = pythonBase(c).
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"hatch", "run", "lint:style"},
+		}).
+		ExitCode(ctx)
+	return err
+}
+
+// Test tests the Python SDK
+func (t Python) Test(ctx context.Context) error {
+	c, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	_, err = pythonBase(c).
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"hatch", "run", "test"},
+		}).
+		ExitCode(ctx)
+	return err
+}
+
+// Generate re-generates the SDK API
+func (t Python) Generate(ctx context.Context) error {
+	c, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	_, err = pythonBase(c).
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"hatch", "run", "generate"},
+		}).
+		ExitCode(ctx)
+	return err
+}
+
+func pythonBase(c *dagger.Client) *dagger.Container {
+	repo := util.Repository(c)
+
+	pyproject := c.Directory().
+		WithFile("pyproject.toml", repo.File("sdk/python/pyproject.toml")).
+		WithFile("README.md", repo.File("sdk/python/README.md"))
+	return c.Container().From("python:3.10-alpine").
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"apk", "add", "-U", "--no-cache", "gcc", "musl-dev", "libffi-dev"},
+		}).
+		WithEnvVariable("PIP_NO_CACHE_DIR", "off").
+		WithEnvVariable("PIP_DISABLE_PIP_VERSION_CHECK", "on").
+		WithEnvVariable("PIP_DEFAULT_TIMEOUT", "100").
+		WithWorkdir("/app/sdk/python").
+		WithMountedDirectory("/app/sdk/python", pyproject).
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"pip", "install", "hatch"},
```

Comment:
I like to add a `"-U", "pip"` on first install to keep it updated. Then no need for the version check disable. It's not only because of that, it's always a good idea to get up to date pip.

Reply:
My concern in doing that is our builds won't be reproducible anymore.

Currently, `pip` is pinned to whichever version of `alpine` we installed. It will not change tomorrow, nor a month from now, until we explicitly decide to change alpine version.